src/applications/find-forms/components/SearchResult.jsx

```diff
@@ -5,63 +5,115 @@ import moment from 'moment';
 import * as customPropTypes from '../prop-types';
 
 // Helper to derive the download link props.
-const deriveLinkProps = form => {
+const deriveLinkPropsFromFormURL = url => {
   const linkProps = {};
+  if (!url) return linkProps;
 
-  const isSameOrigin = form?.attributes?.url.startsWith(window.location.origin);
-  const isPDF = form?.attributes?.url.toLowerCase().includes('.pdf');
+  const isSameOrigin = url.startsWith(window.location.origin);
+  const isPDF = url.toLowerCase().includes('.pdf');
 
   if (!isSameOrigin || !isPDF) {
     // Just open in a new tab if we'd otherwise hit a CORS issue or if the form URL isn't a PDF.
     linkProps.target = '_blank';
   } else {
     // Use HTML5 `download` attribute.
-    linkProps.download = true;
+    linkProps.download = decodeURI(url.split('/').pop());
+    if (isPDF) linkProps.type = 'application/pdf';
   }
 
   return linkProps;
 };
 
+// builds results title
+const buildFormTitle = (id, detailsUrl, title) => (
```

Comment:
Since this function returns JSX it's really a component like anything else so I think we could rename it to just `FormTitle` (or `SearchResultTitle`, or something) and possibly host it in its own file. What do you think?

```jsx
// components/FormTitle.jsx
const FormTitle = ({ id, detailsUrl, title }) => {
   // ...
}

// -> <FormTitle id={id} detailsUrl={detailsUrl} title={title}/>
```